librsvg 2.56.3

skupr-anaconda opened the pull request:
**Destination channel:** main

### Links

- [PKG-6443](https://anaconda.atlassian.net/browse/PKG-6443) 
- [Upstream repository](https://gitlab.gnome.org/GNOME/librsvg)
- Compiling: https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.56.3/INSTALL

### Explanation of changes:

- Add fixes to `bld.bat` and `build.sh` (from the conda-forge recipe)
- Use `vs2022` compiler on Windows
- Rebase a patch
- Update the rust compiler version to 1.82.0 on Windows
- Set `setuptools <74` in `build` to fix the issue with `gobject-introspection` (it has a runtime dependency `setuptools`). distutils.msvccompiler removed in recent setuptools
- Pin `libdeflate` to `1.22`

### Notes:

- `librsvg >=2.56.4` was heavily rewritten and requires `cairo` **1.17.0** but our current `cairo` is **1.16.0**. That's why I'm updating to v**2.56.3**
- `imagecodecs 2024.9.22` requires `libdeflate 1.22`  https://github.com/AnacondaRecipes/imagecodecs-feedstock/pull/17

[PKG-6443]: https://anaconda.atlassian.net/browse/PKG-6443?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

skupr-anaconda committed:
Rebuild with libdeflate 1.22

skupr-anaconda committed:
Add setuptools <65

skupr-anaconda committed:
Add setuptools <65

skupr-anaconda committed:
use python 3.10

skupr-anaconda committed:
Debug g-ir-scanner output

skupr-anaconda committed:
Debugging

skupr-anaconda committed:
Debug

skupr-anaconda committed:
Debug

skupr-anaconda committed:
Use setuptools <74

skupr-anaconda committed:
Use vs2022

skupr-anaconda committed:
Use vs2022

skupr-anaconda committed:
Use vs2022

skupr-anaconda committed:
Add a patch

skupr-anaconda committed:
Add a patch

skupr-anaconda committed:
Add a patch

skupr-anaconda committed:
Add a patch

skupr-anaconda committed:
Add a patch

skupr-anaconda committed:
Revert rust version

skupr-anaconda committed:
2.58.4

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps

ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_06o0ig0054/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:44: should_use_compilers: The recipe requires a compiler directly
===== Final Report: =====
1 Error and 1 Warning were found

</details>

skupr-anaconda committed:
Refresh a patch

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps

ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_e3dt6alz5m/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:44: should_use_compilers: The recipe requires a compiler directly
===== Final Report: =====
1 Error and 1 Warning were found

</details>

skupr-anaconda commented:
```

Requested 'cairo >= 1.17.0' but version of cairo is 1.16.0

Requested 'cairo-png >= 1.17.0' but version of cairo-png is 1.16.0

Requested 'cairo-gobject >= 1.17.0' but version of cairo-gobject is 1.16.0
```

skupr-anaconda committed:
2.56.93

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps

ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_89jlqogaxr/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:44: should_use_compilers: The recipe requires a compiler directly
===== Final Report: =====
1 Error and 1 Warning were found

</details>

skupr-anaconda committed:
2.56.3

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps

ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_d9eg_700db/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:44: should_use_compilers: The recipe requires a compiler directly
===== Final Report: =====
1 Error and 1 Warning were found

</details>

skupr-anaconda committed:
Clean up the recipe

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps

ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_ff_59de2ga/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:77: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:44: should_use_compilers: The recipe requires a compiler directly
===== Final Report: =====
1 Error and 1 Warning were found

</details>

cbouss commented on recipe/meta.yaml:
```diff
@@ -39,8 +39,8 @@ requirements:
     - gobject-introspection
-    - rust_win-64 1.64.0       # [win]
-    - rust        1.64.0       # [win]
+    - rust_win-64 1.82.0       # [win]
+    - rust        1.82.0       # [win]
     - {{ compiler('rust') }}   # [not win]
```
```suggestion
    - {{ compiler('rust') }}
```

skupr-anaconda replied:
Done

cbouss commented on recipe/meta.yaml:
```diff
@@ -76,7 +77,7 @@ requirements:
     - pango
```
```suggestion
    - pango 1.50.7
```

skupr-anaconda replied:
Done

cbouss commented on recipe/meta.yaml:
```diff
@@ -76,7 +77,7 @@ requirements:
     - libzip 1.8.0
-    - libdeflate 1.17
+    - libdeflate 1.22
     - zlib
```
```suggestion
    - zlib {{ zlib }}
```

skupr-anaconda replied:
Done

cbouss commented on recipe/meta.yaml:
```diff
@@ -76,7 +77,7 @@ requirements:
     - pango
```
libiconv run_exports is not applied. Not sure why, but we should at least set pin_compatible in the run section. 
see https://staging.continuum.io/package/prefect/fs/librsvg-feedstock/pr9/75f8f18/osx-64/librsvg-2.56.3-hf076d44_0.tar.bz2

skupr-anaconda replied:
Yeah, that's weird. I added `{{ pin_compatible('libiconv') }}` with a comment.
I have a feeling that all `libiconv` downstreams have the same problem.

Fixed.
Before: `"libiconv",` https://staging.continuum.io/package/prefect/fs/librsvg-feedstock/pr9/0a2fe7a/osx-arm64/librsvg-2.56.3-h44e7d58_0.tar.bz2
After:   `"libiconv >=1.16,<2.0a0"`, https://staging.continuum.io/package/prefect/fs/librsvg-feedstock/pr9/4fbd1ea/osx-arm64/librsvg-2.56.3-h44e7d58_0.tar.bz2

cbouss reviewed (commented):
A few libraries marked as not used on each platforms, including libdeflate.

skupr-anaconda committed:
Update host pinnings, fix rust compiler in build

skupr-anaconda committed:
Set pin_compatible for libiconv in run

skupr-anaconda committed:
Remove libdeflate and zlib from host and run as the library relies on rust crates

skupr-anaconda commented:
> A few libraries marked as not used on each platforms, including libdeflate.

@cbouss I removed `libdeflate` and `zlib` from `host` and `run` because `librsvg` relies on the rust crate `fdeflate`, see https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.56.3/Cargo.lock?ref_type=tags#L688 and on `gio` https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.56.3/Cargo.lock?ref_type=tags#L862 which provides `ZlibDecompressor`, see https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.56.3/src/xml/mod.rs#L4-6.


UPD. Hmm, an error on osx-arm64:
`ERROR (librsvg,bin/rsvg-convert): .. but ['defaults/osx-arm64::zlib==1.2.13=h18a0788_1'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)`

I removed all **optional** dependencies from the specified platforms, see https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.56.3/Cargo.toml#L18-45

skupr-anaconda committed:
Use zlib on osx

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps
WARNING:root:Failed to render using RecipeParserDeps

ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_7c_lfsaqoh/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:91: cbc_dep_in_run_missing_from_host: Run dependencies listed in the cbc should also be present in the host section.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

skupr-anaconda committed:
Do not use optional deps on win

ifitchet commented on recipe/meta.yaml:
```diff
@@ -66,30 +65,30 @@ requirements:
+    - harfbuzz {{ harfbuzz }}      # [not win]
+    - fontconfig {{ fontconfig }}  # [not win]
+    - freetype {{ freetype }}      # [not win]
+    - libiconv 1.16                # [osx]
```
`pango 1.50` is mentioned in `rsvg/Cargo.toml` but I can't see any reference to `libiconv`.  Where is its pinning coming from?

skupr-anaconda replied:
I removed it. Thanks

ifitchet replied:
There's a `pin_compatible` value in `requirements.run` as well...

skupr-anaconda replied:
Well, not that easy. It fails on `osx` without `libiconv`:
```
  ERROR (librsvg,bin/rsvg-convert): Needed DSO lib/libiconv.2.dylib found in ['defaults/osx-arm64::libiconv==1.16=h80987f9_3']

  ERROR (librsvg,bin/rsvg-convert): .. but ['defaults/osx-arm64::libiconv==1.16=h80987f9_3'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

skupr-anaconda replied:
I returned it. I'm not sure why it's happening. Is it glib, fontconfig, or libxml2 can cause it?

ifitchet replied:
Looks like an extra `libiconv` on `osx-64` for `libxml2` (at least):
```
$CONDA_SUBDIR=osx-64 conda search libxml2=2.13.5 --info
...
dependencies: 
  - icu >=73.1,<74.0a0
  - libiconv >=1.16,<2.0a0
  - xz >=5.4.6,<6.0a0
  - zlib >=1.2.13,<1.3.0a0
```

```
$CONDA_SUBDIR=linux-64 conda search libxml2=2.13.5 --info
...
dependencies: 
  - icu >=73.1,<74.0a0
  - libgcc-ng >=11.2.0
  - xz >=5.4.6,<6.0a0
  - zlib >=1.2.13,<1.3.0a0
```

lorepirri replied:
Why `libxml2` does not pull in its "own" `libiconv` then?
https://github.com/AnacondaRecipes/libxml2-feedstock/blob/master/recipe/meta.yaml

skupr-anaconda replied:
@cbouss and I have noticed some issues with `libiconv`'s `run_exports` on osx and win sometimes. Should we update `libiconv` to **1.17** in the next sprints?

skupr-anaconda committed:
Remove libiconv

ifitchet commented:
I don't like

```
export PKG_CONFIG_PATH=${PKG_CONFIG_PATH:-}:${PREFIX}/lib/pkgconfig:${PKG_CONFIG_PATH_FOR_BUILD}:$BUILD_PREFIX/$BUILD/sysroot/usr/lib64/pkgconfig:$BUILD_PREFIX/$BUILD/sysroot/usr/share/pkgconfig
```

as if `PKG_CONFIG_PATH` is not set you'll get `:${PREFIX}/lib/pkgconfig:...`, ie an empty PATH component at the start which might cause issues.

A more correct incantation would be:

```
export PKG_CONFIG_PATH=${PKG_CONFIG_PATH:+${PKG_CONFIG_PATH}:}...
```

Notice the `:` has come inside the braces as well as you only want the colon if you substituted `${PKG_CONFIG_PATH}`.

The same is true for `XDG_DATA_DIRS` higher up.

If you don't like Bash's "Alternate Value" syntax you could say:

```
if [[ ${PKG_CONFIG_PATH} ]] ; then
    PKG_CONFIG_PATH=${PKG_CONFIG_PATH}:...
else
    PKG_CONFIG_PATH=...
fi
```

Noting that the difference is `${PKG_CONFIG_PATH}:` (note the colon) which is exactly the Alternate Value above 😄

ifitchet commented:
Similarly, I'm not quite sure what

```
where=$(which "install" 2>/dev/null || true)

if [ -n "${where}" ]; then
  ln -s -f ${where} ./install
fi

export INSTALL="install"
```

is hoping to achieve as `INSTALL` is not `./install` and so, unless you have `.` on your `PATH` (which is asking for trouble), it's not going to be found when something later runs `$INSTALL` and you will run whatever `which install` gave you in the first place.

ifitchet commented:
Actually, `XDG_DATA_DIRS` is set twice, once unwisely and the second time the non-"Alternate Value" way.

skupr-anaconda committed:
Remove obsolete code

skupr-anaconda commented:
@ifitchet I removed the obsolete code from build.sh. I've checked the conda-forge recipe https://github.com/conda-forge/librsvg-feedstock/blob/main/recipe/build.sh

skupr-anaconda committed:
Remove libiconv from run

skupr-anaconda committed:
Add libiconv on osx

skupr-anaconda committed:
Check the downstream packages

skupr-anaconda commented:
`downstreams` failed:
```
WARNING: failed to get package records, retrying.  exception was: Unsatisfiable dependencies for platform osx-arm64: {MatchSpec("ffmpeg==6.1.1=h4beb5bb_0"), MatchSpec("libxml2[version='>=2.10.4,<2.11.0a0']")}
	chevron_rightTo Task Run

Encountered problems while solving:

  - package ffmpeg-6.1.1-h4beb5bb_0 requires libxml2 >=2.10.4,<2.11.0a0, but none of the providers can be installed
```

We have to rebuild `ffmpeg 6.1.1` with `librsvg 2.56.3`, `libxml2 2.13`

skupr-anaconda committed:
Disable downstream ffmpeg

skupr-anaconda committed:
Disable downstreams

ifitchet reviewed (approved):
I'd remove the downstreams from the recipe completely.

skupr-anaconda committed:
Remove downstreams

skupr-anaconda commented:
> I'd remove the downstreams from the recipe completely.

Done

skupr-anaconda committed:
Remove staging channels
